### PR TITLE
Added indexOf to stringUtils.sol

### PR DESCRIPTION
**Pull request by fivedogit**
This is my ***first ever*** code contribution to an open source project. Be gentle...

### CODE REVIEW

File: library/stringUtils.sol [hunk 1]
```diff
@@ -24,4 +24,32 @@ library StringUtils {
     function equal(string _a, string _b) returns (bool) {
         return compare(_a, _b) == 0;
     }
+    /// @dev Finds the first occurrence of _b in _a and returns its index or -1 if failure
+    function indexOf(string _a, string _b) returns (int) // _a = string to search, _b = string we want to find
```
Comment (chriseth) [inline library/stringUtils.sol]: Some libraries use the names "_haystack" and "_needle" for those variables, which makes the comment mostly unnecessary.

File: library/stringUtils.sol [hunk 2]
```diff
@@ -24,4 +24,32 @@ library StringUtils {
...
+    {
+    	bytes memory a = bytes(_a);
+    	bytes memory b = bytes(_b);
+    	if(a.length < 1 || b.length < 1 || (b.length > a.length))
```
Comment (chriseth) [inline library/stringUtils.sol]: To conform with JavaScript's `indexOf`, we should return 0 for b == ''.

**Comment by chriseth** [comment]
Thanks a lot for the pull request!
It would be great if you could make some small adjustments.
In the long run, we should implement something like Knuth-Morris-Pratt or Boyer-Moore.

File: library/stringUtils.sol (hunk 1 above)
Comment (fivedogit) [inline library/stringUtils.sol]: Oh ok. Cool. I can do that.

File: library/stringUtils.sol (hunk 2 above)
Comment (fivedogit) [inline library/stringUtils.sol]: Ok.

Comment (fivedogit) [inline library/stringUtils.sol]: Actually I just tried if (haystack == needle) return 0, but the compiler says strings and bytes memory can't use the == operator.

**Comment by fivedogit** [comment]
Ok I updated my branch. Can you see and merge the changes?

**Pull request merged**